federation/pkg/kubefed/init/init.go

```diff
@@ -229,6 +232,15 @@ func (i *initFederation) Complete(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	if i.options.apiServerNodePortsPort != 0 {
+		if i.options.apiServerServiceType != v1.ServiceTypeNodePort {
+			return fmt.Errorf("%s should be passed only with '%s=NodePort'", apiserverPortFlag, apiserverServiceTypeFlag)
+		}
+	}
+	if i.options.apiServerNodePortsPort < 0 {
```

Comment:
Based on https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport the cluster has a port range configured. In the context of kubefed we do not know what that port range is thus we cannot check the upper limit.

We made sure that the "out of range" error is propagated to the user, see https://github.com/kubernetes/kubernetes/pull/46283/files#diff-e7243d86cc50e209d78ce51d57dad48dR487